canonical/vanilla-framework
# Update Media Object to support new color theme variables

## Done

- Updated Media Object component to support new color theme variable

Please note the code comment at the top of the file: we can no longer use the `vf-url-friendly-color` function for theming the icons, as it cannot interpolate CSS variables during compilation (since they only exist at runtime). Instead, I hard-coded different inline icons with specific colors for light/dark themes (paper will use the light theme colors). I think this works, particularly given this element is intended to be deprecated anyway.

Fixes [WD-11867](https://warthogs.atlassian.net/browse/WD-11867)

## QA

- Open [default](https://vanilla-framework-5187.demos.haus/docs/examples/patterns/media-object/media-object?theme=light), [large](https://vanilla-framework-5187.demos.haus/docs/examples/patterns/media-object/media-object-large?theme=light), and [circle](https://vanilla-framework-5187.demos.haus/docs/examples/patterns/media-object/media-object-circ-img?theme=light) demos
- Ensure component looks as expected across all themes

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


[WD-11867]: https://warthogs.atlassian.net/browse/WD-11867?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
